mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -354,6 +419,130 @@ class GetRangeSizePattern : public ConvertOpToLLVMPattern<Op> {
     rewriter.replaceOp(op, newValue);
   }
 };
+
+template <typename Op>
+class GetLinearIDPattern : public ConvertOpToLLVMPattern<Op> {
+protected:
+  using typename ConvertOpToLLVMPattern<Op>::OpAdaptor;
+  using ConvertOpToLLVMPattern<Op>::ConvertOpToLLVMPattern;
+
+  virtual Value getID(OpBuilder &builder, Location loc, Type ptrTy,
+                      Value thisArg, int32_t index) const = 0;
+  virtual Value getRange(OpBuilder &builder, Location loc, Type ptrTy,
+                         Value thisArg, int32_t index) const = 0;
+
+public:
+  LogicalResult match(Op) const override { return success(); }
```

Comment:
override -> final ?

Reply:
`match` operations are `override` intentionally, as the derived classes might want to override it (see `ItemNoOffsetGetLinearIDPattern`)

Reply:
Sure. I know that if a class is expected to be derived the virtual member functions that are expected to be overriden cannot be final.